go/vt/discovery/healthcheck.go

```diff
@@ -156,13 +155,13 @@ func init() {
 }
 
 func registerDiscoveryFlags(fs *pflag.FlagSet) {
-	fs.StringVar(&TabletURLTemplateString, "tablet_url_template", "http://{{.GetTabletHostPort}}", "format string describing debug tablet url formatting. See the Go code for getTabletDebugURL() how to customize this.")
-	fs.DurationVar(&refreshInterval, "tablet_refresh_interval", 1*time.Minute, "tablet refresh interval")
-	fs.BoolVar(&refreshKnownTablets, "tablet_refresh_known_tablets", true, "tablet refresh reloads the tablet address/port map from topo in case it changes")
-	fs.IntVar(&topoReadConcurrency, "topo_read_concurrency", 32, "concurrent topo reads")
-	fs.StringSliceVar(&tabletFilters, "tablet_filters", []string{}, "Specifies a comma-separated list of 'keyspace|shard_name or keyrange' values to filter the tablets to watch")
-	topoproto.TabletTypeListVar(&AllowedTabletTypes, "allowed_tablet_types", "Specifies the tablet types this vtgate is allowed to route queries to")
-	fs.StringSliceVar(&KeyspacesToWatch, "keyspaces_to_watch", []string{}, "Specifies which keyspaces this vtgate should have access to while routing queries or accessing the vschema")
+	fs.StringVar(&TabletURLTemplateString, "tablet_url_template", "http://{{.GetTabletHostPort}}", "Format string describing debug tablet url formatting. See getTabletDebugURL() for how to customize this.")
+	fs.DurationVar(&refreshInterval, "tablet_refresh_interval", 1*time.Minute, "Tablet refresh interval.")
+	fs.BoolVar(&refreshKnownTablets, "tablet_refresh_known_tablets", true, "Whether to reload the tablet's address/port map from topo in case they change.")
+	fs.IntVar(&topoReadConcurrency, "topo_read_concurrency", 32, "Concurrency of topo reads.")
+	fs.StringSliceVar(&tabletFilters, "tablet_filters", []string{}, "Specifies a comma-separated list of 'keyspace|shard_name or keyrange' values to filter the tablets to watch.")
+	topoproto.TabletTypeListVar(fs, &AllowedTabletTypes, "allowed_tablet_types", "Specifies the tablet types this vtgate is allowed to route queries to.")
```

Comment:
so what we have now is fine (possibly obvious to say 😂 ) but it would be more "conventional" (now that i'm thinking about the other flag to) to have _just_ a type which implements the `pflag.Value` interface (called `topoproto.TabletTypeListFlag` or similar), and then here do:

```go
fs.Var((*topoproto.TabletTypeListFlag)(&AllowedTabletTypes), "allowed_tablet_types", ...)
```

where that's defined as:

```go
type TabletTypeListFlag []topodatapb.TabletType

// attach Value implementation to TabletTypeListFlag
```

Reply:
(I would then do the same for the `TabletTypeVar` (which lets us use it on both `flag` and `pflag` flagsets, as a bonus) in #10861)

Reply:
oh ... 🤦 we already have exactly what i'm describing, so let's just do that.

Reply:
like [this](https://github.com/vitessio/vitess/pull/10867/files#diff-35c7445dcf9c320905f943e3c82d0715767ae4493597db73d49d108cee27a05a) (and if you do that for the list type, we can close 10861, or I can add that to my PR after we merge yours, totally up to you!)

Reply:
`pflag.Value` is essentially `flag.Value` plus that `Type` function, so you can use it with `flag` "for free" (`Get` is vitess's own invention and has nothing to do with `flag.Value`, you can delete it if you don't see it used)